async-nats/src/options.rs

```diff
@@ -175,6 +177,38 @@ impl ConnectOptions {
         crate::connect_with_options(addrs, self).await
     }
 
+    /// Use a builder to specify a custom auth callback to be used when authenticating against the NATS Server.
```

Comment:
Creates a builder with a custom auth callback to be used when ... ?

Reply:
agree.